packages/grid/data-grid/src/tests/pagination.DataGrid.test.tsx

```diff
@@ -187,5 +193,50 @@ describe('<DataGrid /> - Pagination', () => {
       fireEvent.click(screen.getByRole('button', { name: /next page/i }));
       expect(getColumnValues()).to.deep.equal(['Nike 1']);
     });
+
+    describe('AutoPageSize', () => {
+      it('should always render the same amount of rows and fit the viewport', () => {
```

Comment:
The test doesn't seem to be failing on HEAD. Is it testing the bug?

Reply:
It's because I changed the height and the last 2 pages had 3 rows...
Fixed now